app/routes/api.gitlab.description.ts

```diff
@@ -0,0 +1,60 @@
+import { type ActionFunctionArgs, json } from '@remix-run/cloudflare';
+import { GitlabService } from '~/lib/persistenceGitbase/gitlabService';
+import { withV8AuthUser } from '~/lib/verse8/middleware';
+
+export const action = withV8AuthUser(descriptionAction, { checkCredit: true });
+
+/**
+ * Action function for updating project description
+ */
+async function descriptionAction({ context, request }: ActionFunctionArgs) {
+  const env = { ...context.cloudflare.env, ...process.env } as Env;
+  const user = context?.user as { email: string; isActivated: boolean };
+
+  if (!user) {
+    return json({ success: false, message: 'Authentication required' }, { status: 401 });
+  }
+
+  if (!user.isActivated) {
+    return json({ success: false, message: 'User is not activated' }, { status: 403 });
+  }
```

Comment:
We might need to abstract it to aid duplicated codes.

Reply:
Inside withV8AuthUser, it is already being checked, but it seems to be checked again outside. I will remove it